LeetCodeSolutions/PathSumIII/PathSumIII.cs

```diff
@@ -0,0 +1,67 @@
+namespace LeetCodeSolutions;
+public class PathSumIIISolution
+{
+    private class TreeSearcher
+    {
+        private int _numberOfMatchingPaths;
+        private readonly Dictionary<long, int> _frequencyOfSums;
+        private readonly long _targetSum;
+        public TreeSearcher(int targetSum)
+        {
+            _targetSum = targetSum;
+            _numberOfMatchingPaths = 0;
+            //there is always one path of sum zero, the null set so to speak
+            _frequencyOfSums = new()
+            {
+                { 0, 1 }
+            };
+        }
+        public int NumberOfPaths => _numberOfMatchingPaths;
+        private void Lookup(long sum)
```

Comment:
give this a return value so it's clear to the calling level what it's supposed to do